### ISSUE TITLE
Wrong type of ExcerptFunction in Typescript?

### ISSUE THREAD

**Issue by Raynirs** [antfu/vite-plugin-md#88]
ExcerptFunction is defined as
```typescript
declare type ExcerptFunction = (contents: string, options: GraymatterOptions) => string;
```
but accually it returns void and the first parameter is an object as shown in the comment
```typescript
function firstFourLines(file, options) {
   file.excerpt = file.content
     .split('\n')
     .slice(0, 4)
     .join(' ')
}
```
I have to change the type in source code manually to avoid an error. Maybe this can be fixed easily.

**Comment by yankeeinlondon** [comment]
@Raynirs I can look at this if you can put this into a minimally reproducible example on [Stackblitz](https://vite.new/vue)

**Comment by yankeeinlondon** [comment]
fixed in `v0.16.0`; please follow the _type_ strictly and it will work or look at the tests I've added if you're not certain. I will add more to docs later.
